stonesoup/models/measurement/nonlinear.py

```diff
@@ -1400,3 +1401,346 @@ def rvs(self, num_samples=1, **kwargs) -> Union[StateVector, StateVectors]:
+    """
+
+    target_location: StateVector = Property(
+        default=StateVector([0, 0, 0]),
+        doc="A 3x1 array specifying the Cartesian Target location in terms of :math:`x,y,z` "
+            "coordinates.")
+
+    translation_offset: StateVector = Property(
+        default=None,
+        doc="A 3x1 array specifying the Cartesian origin offset in terms of :math:`x,y,z` "
+            "coordinates.")
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        # Set values to defaults if not provided
+        if self.translation_offset is None:
+            self.translation_offset = StateVector([0] * 3)
+        if self.target_location is None:
+            # It should be given
+            self.translation_offset = StateVector([0] * 3)
+
+    @property
+    def ndim_meas(self):
+        return 2
+
+    def function(self, state,
+                 noise=False,
+                 **kwargs) -> StateVectors:
+
+        if isinstance(noise, bool) or noise is None:
+            if noise:
+                noise = self.rvs(num_samples=state.state_vector.shape[1], **kwargs)
+            else:
+                noise = 0
+
+        # keep the case of multiple points
+        npts = state.state_vector.shape[1]
+
+        azimuths = np.zeros((1, npts))
+        elevations = np.zeros((1, npts))
+
+        for ipoint in range(npts):
+            # adjust the sensor location with the translation offset, if present
+            sensor_location = state.state_vector[self.mapping, ipoint] - \
+                              self.translation_offset
+
+            diff_position = np.array([self.target_location -
+                                      sensor_location]).reshape(-1)
+
+            # evaluate the range
+            rg = np.sqrt(diff_position[0] * diff_position[0] +
+                         diff_position[1] * diff_position[1] +
+                         diff_position[2] * diff_position[2])
+
+            # evaluate the absolute azimuth and elevation of the target respect to the sensor
+            absolute_azimuth = np.array(atan2(diff_position[1], diff_position[0])).reshape(-1, 1)
+            absolute_elevation = np.array(asin(diff_position[2] / rg)).reshape(-1, 1)
+
+            # evaluate the sensor heading and elevation
+            heading = np.array(np.radians(state.state_vector[9, ipoint])).reshape(-1, 1)
+            pitch = np.array(np.radians(state.state_vector[11, ipoint])).reshape(-1, 1)
+
+            # Transform the azimuths and elevation and fix for 180 degrees in case
+            azimuths[:, ipoint] = [Azimuth(angle_wrap(angle - heading[index])) for index, angle in
+                                   enumerate(absolute_azimuth)]
+            elevations[:, ipoint] = [Elevation(angle - pitch[index]) for index, angle in
+                                     enumerate(absolute_elevation)]
+
+        return StateVector([*azimuths, *elevations]) + noise
```

Comment:
```suggestion
        # adjust the sensor location with the translation offset, if present
        sensor_location = state.state_vector[self.mapping, :] - self.translation_offset

        diff_position = self.target_location - sensor_location

        # evaluate the range
        rg = np.linalg.norm(diff_position, axis=0)

        # evaluate the absolute azimuth and elevation of the target respect to the sensor
        absolute_azimuth = np.arctan2(diff_position[1, :], diff_position[0, :])
        absolute_elevation = np.arcsin(diff_position[2, :] / rg)

        # evaluate the sensor heading and elevation
        heading = np.radians(state.state_vector[9, :])
        pitch = np.radians(state.state_vector[11, :])

        # Transform the azimuths and elevation and fix for 180 degrees in case
        azimuths = [Azimuth(angle) for angle in absolute_azimuth - heading]
        elevations = [Elevation(angle) for angle in absolute_elevation - pitch]

        return StateVectors([azimuths, elevations]) + noise
```